### PR TITLE
PIM-9608: Merge 4.0 into master

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/InternalApi/AxisValueLabelsNormalizer/MetricNormalizer.php [hunk 1]
```diff
@@ -22,15 +23,26 @@ class MetricNormalizer implements AxisValueLabelsNormalizer
     /** @var MetricLocalizer */
     private $metricLocalizer;
 
-    public function __construct(StandardMetricNormalizer $metricNormalizer, MetricLocalizer $metricLocalizer)
-    {
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /**
+     * @todo @merge master/5.0: replace translator argument by the adequate service
```
Comment (SamirBoulil) [inline src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/InternalApi/AxisValueLabelsNormalizer/MetricNormalizer.php:30]: Todo

File: src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/InternalApi/AxisValueLabelsNormalizer/MetricNormalizer.php [hunk 2]
```diff
@@ -43,16 +55,36 @@ public function normalize(ValueInterface $value, string $locale): string
...
+        );
     }
 
     public function supports(string $attributeType): bool
     {
         return AttributeTypes::METRIC === $attributeType;
     }
+
+    /**
+     * @todo @merge master/5.0: rework this method
```
Comment (SamirBoulil) [inline src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/InternalApi/AxisValueLabelsNormalizer/MetricNormalizer.php:76]: Todo

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Purge/PurgeJobExecution.php [hunk 3]
```diff
@@ -23,18 +24,28 @@ final class PurgeJobExecution
     /** @var DeleteOrphanJobExecutionDirectories */
     private $deleteOrphansJobExecutionDirectories;
 
+    // @todo merge master: remove null
```
Comment (SamirBoulil) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Purge/PurgeJobExecution.php:27]: search for "todo" and fix before merge on master ;)

**Review by SamirBoulil — APPROVED** [review]
GTM after fix

**Review by ValentinMumble — APPROVED** [review]

**Review by ahocquard — APPROVED** [review]

**Pull request merged**